### PR TITLE
use two simple regexes to workaround macos compilers producing endless loops

### PR DESCRIPTION
**Pull request by dothebart**
Older macs seem to produce endless loops with these complex regexes. Break it up into two simple regexes as a workaround.

https://stackoverflow.com/questions/33795759/c-mac-os-x-regex-causes-infinite-loop-with-regex-replace/33799633

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4902/

**Comment by dothebart** [comment]
unrelated build error on one machine - probably overload or machine unstable?

File: lib/ProgramOptions/Parameters.cpp [hunk 1]
```diff
@@ -25,16 +25,20 @@
 #include <regex>
 
 namespace {
-std::regex const removeComments("(^[ \t]+|[ \t]*(#.*)?$)", std::regex::nosubs | std::regex::ECMAScript);
+  std::regex const removeComments("#.*$", std::regex::ECMAScript);
+  std::regex const removeTabs("[ \t]*", std::regex::ECMAScript);
```
Comment (ObiWahn) [inline lib/ProgramOptions/Parameters.cpp]: please add an ^ here

File: lib/ProgramOptions/Parameters.cpp [hunk 2]
```diff
@@ -25,16 +25,20 @@
...
 }
 
 namespace arangodb {
 namespace options {
 
 std::string removeCommentsFromNumber(std::string const& value) {
-  // replace leading spaces, replace trailing spaces & comments
-  return std::regex_replace(value, ::removeComments, "");
+  // replace trailing comments
+  auto noComment = std::regex_replace(value, ::removeComments, "");
```
Comment (ObiWahn) [inline lib/ProgramOptions/Parameters.cpp]: please swap the application order of the 2 operations. Then the `^#` will work correctly.

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-mac-onlycatch/2882/

**Comment by dothebart** [comment]
Tests blue.

File: lib/ProgramOptions/Parameters.cpp [hunk 3]
```diff
@@ -25,16 +25,20 @@
 #include <regex>
 
 namespace {
-std::regex const removeComments("(^[ \t]+|[ \t]*(#.*)?$)", std::regex::nosubs | std::regex::ECMAScript);
+  std::regex const removeComments("#.*$", std::regex::ECMAScript);
+  std::regex const removeTabs("^[ \t]*|[ \t]*$", std::regex::ECMAScript);
```
Comment (jsteemann) [inline lib/ProgramOptions/Parameters.cpp]:
```suggestion
  std::regex const removeTabs("^[ \t]+|[ \t]+$", std::regex::ECMAScript);
```

File: lib/ProgramOptions/Parameters.cpp [hunk 4]
```diff
@@ -25,16 +25,20 @@
...
 namespace arangodb {
 namespace options {
 
 std::string removeCommentsFromNumber(std::string const& value) {
-  // replace leading spaces, replace trailing spaces & comments
-  return std::regex_replace(value, ::removeComments, "");
+  // replace trailing comments
+  auto noComment = std::regex_replace(value, ::removeComments, "");
+  // replace leading spaces, replace trailing spaces
+  auto plainValue = std::regex_replace(noComment, ::removeTabs, "", std::regex_constants::match_not_null);
```
Comment (jsteemann) [inline lib/ProgramOptions/Parameters.cpp]:
```suggestion
  return std::regex_replace(noComment, ::removeTabs, "");
```

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4939/

**Comment by dothebart** [comment]
tests blue.

**Pull request merged**